### ISSUE TITLE
Error 1163 being triggered for Parameters required by interface.

### ISSUE THREAD

**Issue by SamuelEnglard** [JosefPihrt/Roslynator#48]
I have parameters that I'm not using but are required by the interface the method is implementing. RCS1163 should not fire in these cases.

**Comment by John-Welch** [comment]
It's similarly fired by event handlers that aren't using 'sender' or 'e' parameters.

**Comment by JosefPihrt** [comment]
Could you please provide a code sample?

**Comment by SamuelEnglard** [comment]
```csharp
class A : ICompare
{
    int Compare(object obj) => 0;
}
```
`object obj` is required for `ICompare` but will trigger 1163 (Admittedly this is a stupid case but should still work)

**Comment by JosefPihrt** [comment]
I tried your sample but it did not trigger RCS1163.

**Comment by John-Welch** [comment]
While I first noticed the RCS1163 reports at work (on some code I can't post), it's also seen on some (ancient) code I have sitting around at home:

`	public class MandelbrotPanel: Control
	{
...
		private BackgroundWorker bitmapMaker = new BackgroundWorker();

		public MandelbrotPanel()
		{
			bitmapMaker.DoWork += bitmapMaker_DoWork;
			bitmapMaker.ProgressChanged += bitmapMaker_ProgressChanged;
			bitmapMaker.RunWorkerCompleted += bitmapMaker_RunWorkerCompleted;
		}

		void bitmapMaker_RunWorkerCompleted(object sender, RunWorkerCompletedEventArgs e)
// Reports RCS1163 Unused parameter 'sender'
		{
			if (e.Cancelled)
			{
				BeginBitmapPainting();
			}
		}

		void bitmapMaker_ProgressChanged(object sender, ProgressChangedEventArgs e)
// Reports RCS1163 Unused parameter 'sender'
// Reports RCS1163 Unused parameter 'e'
		{
			BitmapData bmd = bitmap.LockBits(new Rectangle(0, 0, bitmap.Width, bitmap.Height),
				ImageLockMode.WriteOnly, PixelFormat.Format24bppRgb);
			try
			{
				lock (bitmapDataTransferLock)
				{
					System.Runtime.InteropServices.Marshal.Copy(bitmapDataTransfer, 0, bmd.Scan0, bitmapDataTransfer.Length);
				}
			}
			finally
			{
				bitmap.UnlockBits(bmd);
			}
			Invalidate();
		}
	}
`

**Comment by ThorstenHouy** [comment]
I can confirm this. In my app it is firing for every Event.

Example:
TaskScheduler.UnobservedTaskException+= TaskScheduler_UnobservedTaskException;

// Reports RCS1163 Unused parameter 'sender'
// Reports RCS1163 Unused parameter 'e'
private void TaskScheduler_UnobservedTaskException(object sender, UnobservedTaskExceptionEventArgs e)
{

}

![unusedparameter](https://cloud.githubusercontent.com/assets/25899448/23156110/e747d7e4-f815-11e6-8e87-9fc7516b6145.jpg)

**Comment by JosefPihrt** [comment]
I agree that event handler methods should not be reported. I moved this case to #49.

**Comment by JosefPihrt** [comment]
Is anybody experiencing a behaviour that RCS1163 is triggered by a parameter of a method that implements interface?

**Comment by SamuelEnglard** [comment]
I can't create a repo that's not my product code yet ☹️

**Comment by John-Welch** [comment]
This example shows Interface methods having RCS1163 for all the method parameters:
```C#
   class Dummy<T> : ICollection<T>
    {
        public int Count
        {
            get { throw new NotSupportedException(); }
        }

        public bool IsReadOnly
        {
            get { throw new NotSupportedException(); }
        }

        public void Add(T item)
        {
            throw new NotSupportedException();
        }

        public void Clear()
        {
            throw new NotSupportedException();
        }

        public bool Contains(T item)
        {
            throw new NotSupportedException();
        }

        public void CopyTo(T[] array, int arrayIndex)
        {
            throw new NotSupportedException();
        }

        public IEnumerator<T> GetEnumerator()
        {
            throw new NotSupportedException();
        }

        public bool Remove(T item)
        {
            throw new NotSupportedException();
        }

        IEnumerator IEnumerable.GetEnumerator()
        {
            throw new NotSupportedException();
        }
    }
```

**Comment by JosefPihrt** [comment]
I was confused because I've fixed this bug meanwhile, without realizing that I'm actually fixing it. So I was unable to reproduce it on master.

**Comment by JosefPihrt** [comment]
Fixed with a6b2e46b22e9a570fb352e867ed2f753aefa827f
